infra/indexers/src/usdc-balances.indexer.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  Block,
+  FieldElement,
+  Filter,
+  formatUnits,
+  hash,
+  uint256,
+} from "./deps.ts";
+import { balanceStorageLocation, USDC_ADDRESS, USDC_DECIMALS } from "./usdc.ts";
+
+const filter: Filter = {
+  header: {
+    weak: true,
+  },
+  events: [
+    {
+      fromAddress: USDC_ADDRESS,
+      keys: [hash.getSelectorFromName("Transfer") as FieldElement],
+      includeReceipt: false,
+    },
+  ],
+  stateUpdate: {
+    storageDiffs: [
+      { contractAddress: USDC_ADDRESS },
+    ],
+  },
+};
+
+export const config = {
+  streamUrl: "https://mainnet.starknet.a5a.ch",
+  startingBlock: 274621,
+  network: "starknet",
+  finality: "DATA_STATUS_ACCEPTED",
+  filter,
+  sinkType: "postgres",
+  sinkOptions: {
+    tableName: "balance_usdc",
+  },
+};
+
+export default function decodeUSDCBalances(
+  { header, events, stateUpdate }: Block,
+) {
+  const { blockNumber, timestamp } = header!;
+
+  // Step 1: collect addresses that have been part of a transfer.
+  const addresses = (events ?? []).reduce((addresses, { event }) => {
+    const [fromAddress, toAddress] = event.data;
+    addresses.add(fromAddress);
+    addresses.add(toAddress);
+    return addresses;
+  }, new Set<string>());
+
+  // Step 2: collect balances for each address.
+  const storageDiffs = stateUpdate?.stateDiff?.storageDiffs ?? [];
+  if (storageDiffs.length !== 1) {
```

Comment:
can never be 2 ? we are working with u256 here no ?

Reply:
Storage diffs are grouped by contract, since we request storage diffs for usdc only it should always be 1. This condition is mostly a sanity check.

Reply:
hmm as it's a transfer we should have 2 then ? sender + recipient ?

Reply:
ahh gotcha